doc/designs/passkeys.md

```diff
@@ -0,0 +1,246 @@
+# Passkey authentication
+
+## Overview
+
+Traditional authentication with a password is not considered secure enough
+by many companies or government agencies. Alternate and more secure
+solutions exist, among which the use of passkeys, where the private
+key is stored on the device and the server only needs to know the public
+key.
+
+The goal of this feature is to use a passkey to authenticate a user
+against IPA.
+
+The project will be jointly developed by SSSD and IPA:
+
+- IPA provides the interface to store the user's public credentials
+- IPA provides the interface to configure passkey settings
+- SSSD performs the actual authentication
+
+## Use Cases
+
+- The administrator or the user registers a passkey into IPA, associated
+to a user account. The registration requires the insertion of the key, or
+the string output provided by SSSD registration mechanism.
```

Comment:
Please change the sentence to something like "The registration process stores a description of the passkey bound to IPA deployment and requires a direct communication with the passkey device. Alternatively the description string can be obtained through the SSSD registration tool and added without the presence of the passkey device."